sormas-backend/src/main/resources/sql/sormas_schema.sql

```diff
@@ -7940,4 +7940,329 @@ ALTER TABLE task_history ADD COLUMN travelentry_id bigint;
+                            rec.vaccineuniicode, rec.vaccineatccode, rec.vaccinationinfosource,
+                            rec.pregnant, rec.trimester
+                        );
+                END IF;
+            END LOOP;
+    END;
+$$ LANGUAGE plpgsql;
+
+DROP TABLE IF EXISTS tmp_vaccinated_entities;
+DROP TABLE IF EXISTS tmp_vaccinated_persons;
+DROP FUNCTION IF EXISTS clone_healthconditions(bigint);
+DROP FUNCTION IF EXISTS create_healthconditions();
+DROP FUNCTION IF EXISTS create_vaccination(bigint, bigint, timestamp, bigint, timestamp, varchar(255), text, varchar(255), text, text, text, text, text, varchar(255), varchar(255), varchar(255));
+
+/* Step 4: Clean up cases, contacts and event participants */
+ALTER TABLE cases RENAME COLUMN vaccination TO vaccinationstatus;
+ALTER TABLE cases_history RENAME COLUMN vaccination TO vaccinationstatus;
+ALTER TABLE cases RENAME COLUMN lastvaccinationdate TO smallpoxlastvaccinationdate;
+ALTER TABLE cases_history RENAME COLUMN lastvaccinationdate TO smallpoxlastvaccinationdate;
+UPDATE cases SET smallpoxlastvaccinationdate = null WHERE disease != 'MONKEYPOX';
+
+ALTER TABLE cases DROP COLUMN vaccinationdoses;
+ALTER TABLE cases DROP COLUMN vaccinationinfosource;
+ALTER TABLE cases DROP COLUMN firstvaccinationdate;
+ALTER TABLE cases DROP COLUMN vaccinename;
+ALTER TABLE cases DROP COLUMN othervaccinename;
+ALTER TABLE cases DROP COLUMN vaccinemanufacturer;
+ALTER TABLE cases DROP COLUMN othervaccinemanufacturer;
+ALTER TABLE cases DROP COLUMN vaccineinn;
+ALTER TABLE cases DROP COLUMN vaccinebatchnumber;
+ALTER TABLE cases DROP COLUMN vaccineuniicode;
+ALTER TABLE cases DROP COLUMN vaccineatccode;
+ALTER TABLE cases DROP COLUMN vaccine;
+
+ALTER TABLE contact ADD COLUMN vaccinationstatus varchar(255);
+ALTER TABLE contact_history ADD COLUMN vaccinationstatus varchar(255);
+ALTER TABLE eventparticipant ADD COLUMN vaccinationstatus varchar(255);
+ALTER TABLE eventparticipant_history ADD COLUMN vaccinationstatus varchar(255);
+
+UPDATE contact SET vaccinationstatus = vaccinationinfo.vaccination, changedate = now() FROM vaccinationinfo WHERE contact.vaccinationinfo_id = vaccinationinfo.id;
+UPDATE eventparticipant SET vaccinationstatus = vaccinationinfo.vaccination, changedate = now() FROM vaccinationinfo WHERE eventparticipant.vaccinationinfo_id = vaccinationinfo.id;
+
+ALTER TABLE contact DROP COLUMN vaccinationinfo_id;
+ALTER TABLE eventparticipant DROP COLUMN vaccinationinfo_id;
+DROP TABLE IF EXISTS vaccinationinfo;
+
+UPDATE exportconfiguration SET propertiesstring = replace(propertiesstring, 'vaccination,', 'vaccinationstatus,');
+
+UPDATE featureconfiguration SET enabled = true WHERE featuretype = 'IMMUNIZATION_MANAGEMENT';
+UPDATE featureconfiguration SET enabled = true WHERE featuretype = 'IMMUNIZATION_STATUS_AUTOMATION';
+/* End of vaccination refactoring */
+
+INSERT INTO schema_version (version_number, comment) VALUES (405, 'Vaccination refactoring #5909');
```

Comment:
We have to keep the history table columns anyway; basically we can never drop history table columns, even if we simply drop the column from the main table (and don't migrate it anywhere) because we need to be able to access that data if it's requested later.